helper/helper_test.go

```diff
@@ -1,12 +1,203 @@
+	Expect(h.currentStep).To(BeNumerically("<", len(h.expectedSteps)))
+	return h.expectedSteps[h.currentStep]
+}
+
+func (h *restoreMockHelperImpl) closeAndDeletePipe(tableOid int, batchNum int) {
+	pipename := fmt.Sprintf("%s_%d_%d", *pipeFile, tableOid, batchNum)
+	Expect(h.isPipeCreated(pipename)).To(BeTrue())
+
+	delete(h.pipesOpened, pipename)
+	delete(h.pipesCreated, pipename)
+}
+
+func newHelperTest(batches []oidWithBatch, steps []helperTestStep) *restoreMockHelperImpl {
+	var ret = new(restoreMockHelperImpl)
+	ret.expectedOidBatch = batches
+	ret.expectedSteps = steps
+	ret.pipesCreated = make(map[string]bool)
+	ret.pipesOpened = make(map[string]bool)
+	ret.restoreData = &restoreReaderTestImpl{}
+
+	// pre-create pipes as starter does
+	for i := 0; i < *copyQueue; i++ {
+		oidBatch := ret.expectedOidBatch[i]
+		pipename := fmt.Sprintf("%s_%d_%d", *pipeFile, oidBatch.oid, oidBatch.batch)
+		ret.createPipe(pipename)
+	}
+	return ret
+}
+
+func (h *restoreMockHelperImpl) getOidWithBatchListFromFile(oidFileName string) ([]oidWithBatch, error) {
+	return h.expectedOidBatch, nil
+}
+
+func (h *restoreMockHelperImpl) checkForSkipFile(pipeFile string, tableOid int) bool {
+	step := h.getCurStep()
+	Expect(tableOid).To(Equal(step.skipFileArgTableOid))
+	ret := step.skipFileResult
+	return ret
+}
+
+func (h *restoreMockHelperImpl) createPipe(pipe string) error {
+	// Check that pipe was not opened yet
+	Expect(h.isPipeCreated(pipe)).To(Equal(false))
+	Expect(h.isPipeOpened(pipe)).To(Equal(false))
+
+	h.pipesCreated[pipe] = true
+	return nil
+}
+
+func (h *restoreMockHelperImpl) flushAndCloseRestoreWriter(pipeName string, oid int) error {
+	// Check that we are closing pipe which is opened
+	Expect(h.isPipeOpened(pipeName)).To(Equal(true))
+	delete(h.pipesOpened, pipeName)
+	return nil
+}
+
+func (*restoreMockHelperImpl) doRestoreAgentCleanup() {
+	// This was intentionaly left blank to support the IRestoreHelper interface
+}
+
+func (h *restoreMockHelperImpl) getRestoreDataReader(fileToRead string, objToc *toc.SegmentTOC, oidList []int) (IRestoreReader, error) {
+	Expect(h.restoreData).ToNot(BeNil())
+	return h.restoreData, nil
+}
+
+func (h *restoreMockHelperImpl) getRestorePipeWriter(currentPipe string) (*bufio.Writer, *os.File, error) {
+	step := h.makeStep()
+	Expect(currentPipe).To(Equal(step.restorePipeWriterArgExpect))
+
+	// The pipe is opened in getRestorePipeWriter and should not be created before
+	Expect(h.isPipeOpened(currentPipe)).To(Equal(false))
+
+	if step.restorePipeWriterResult {
+		h.pipesOpened[currentPipe] = true
+		var writer bufio.Writer
+		return &writer, nil, nil
+	}
+	return nil, nil, unix.ENXIO
+}
+
+type testPluginCmd struct {
+	hasProcess_ bool
+	error       *string
+	waitCount   int
+}
+
+func (tp *testPluginCmd) hasProcess() bool {
+	return tp.hasProcess_
+}
+
+func (pt *testPluginCmd) pid() int {
+	return 42
+}
+
+func (pt *testPluginCmd) Wait() error {
+	pt.waitCount += 1
```

Comment:
```suggestion
	pt.waitCount++
```

Reply:
What is the reason for this change? The code won't be faster, cleaner, more readable, etc. Then why?

Reply:
because `currentStep++`

Reply:
still don't get the reason

Reply:
unification